From f60c33b5afc4b1dff0b31f17d7ae222db8aa789f Mon Sep 17 00:00:00 2001
From: Marek Chalupa <mchqwerty@gmail.com>
Date: Fri, 3 Jul 2015 11:28:00 +0200
Subject: frames: handle META_FRAME_CONTROL_NONE on left click

We can get this operation in some cases, for example when
we're trying to resize window that cannot be resized.
This can occur with maximized windows that have a border
(without border we couldn't resize them by mouse in maximized state).
In this case we reached abort() beacuse we did not handle this op.

https://bugzilla.gnome.org/show_bug.cgi?id=751884

diff --git a/src/ui/frames.c b/src/ui/frames.c
index 362d7b6..a2f7f45 100644
--- a/src/ui/frames.c
+++ b/src/ui/frames.c
@@ -1053,6 +1053,11 @@ meta_frame_left_click_event (MetaUIFrame *frame,
       }
 
       return TRUE;
+    case META_FRAME_CONTROL_NONE:
+      /* We can get this for example when trying to resize window
+       * that cannot be resized (e. g. it is maximized and the theme
+       * currently used has borders for maximized windows), see #751884 */
+      return FALSE;
     default:
       g_assert_not_reached ();
     }
-- 
cgit v0.10.2

